yt_dlp/extractor/afreecatv.py

```diff
@@ -380,3 +383,177 @@ def _real_extract(self, url):
+                    errnote=f'Unable to download access token for {quality_str} stream',
+                )
+                aid = aid_response['CHANNEL']['AID']
+
+                stream_info_url = (
+                    channel_info.get('RMD', 'https://livestream-manager.afreecatv.com')
+                    + self._STREAM_API_PATH
+                )
+                cdn = channel_info.get('CDN', self._CDNS[0])
+                stream_key = f'{broadcast_no}-{stream_key_type}-{quality_str}-hls'
+                stream_info = self._download_json(
+                    stream_info_url,
+                    broadcast_no,
+                    query={
+                        'return_type': cdn,
+                        'use_cors': True,
+                        'cors_origin_url': 'play.afreecatv.com',
+                        'broad_key': stream_key,
+                        # TODO the website passes a "time" parameter, but it's not obvious how it's derived
+                        # for now we can get away with ignoring it
+                    },
+                    note=f'Downloading metadata for {quality_str} stream',
+                    errnote=f'Unable to download metadata for {quality_str} stream',
+                    fatal=False,
+                )
+                view_url = stream_info['view_url']
+
+                formats.append({
+                    'format_id': quality_str,
+                    'quality': quality_key(quality_str),
+                    'ext': 'mp4',
+                    'protocol': 'm3u8',
+                    'url': update_url_query(view_url, {
+                        'aid': aid,
+                    }),
+                })
+            except (ExtractorError, KeyError):
+                continue
+
+        self._sort_formats(formats)
+        info = {
+            'id': broadcast_no,
+            'title': channel_info.get('TITLE'),
+            'uploader': channel_info.get('BJNICK'),
+            'uploader_id': channel_info.get('BJID'),
+            'formats': formats,
+        }
+
+        # try for a little bit of extra metadata
+        try:
+            station_info = self._download_json(
+                self._STATION_API_URL,
+                broadcaster_id,
+                query={
+                    'szBjId': broadcaster_id,
+                },
+                note='Downloading channel metadata',
+                errnote='Unable to download channel metadata',
+            )
+            info['upload_date'] = unified_strdate(station_info.get('broad_start'))
+
+            if not info['title']:
+                # not exactly equivalent to stream title, use only as fallback
+                info['title'] = station_info.get('station_title')
+            if not info['uploader']:
+                info['uploader'] = station_info.get('station_name')
+        except Exception as e:
+            self.report_warning('Failed to look up extra metadata: ' + error_to_compat_str(e))
```

Comment:
no need for this to be in try-except block.
Just let it be `None` if extraction failed

Reply:
Remove all these and then inline the info dict